exporters/common/src/main/java/io/opentelemetry/exporter/internal/marshal/Serializer.java

```diff
@@ -122,14 +122,27 @@ public void serializeSInt32(ProtoFieldInfo field, int value) throws IOException
 
   protected abstract void writeSInt32(ProtoFieldInfo info, int value) throws IOException;
 
-  /** Serializes a protobuf {@code uint32} field. */
+  /** Serializes a protobuf {@code int32} field. */
   public void serializeInt32(ProtoFieldInfo field, int value) throws IOException {
     if (value == 0) {
       return;
     }
     writeint32(field, value);
   }
 
+  /** Serializes a protobuf {@code int32} field. */
```

Comment:
Nit:  I'd explicitly call this out as serialization a `{@code optional int32} field.`

Yes, it's in the name of the method.